src/triage/component/collate/from_obj.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from triage.validation_primitives import (
+    table_should_exist,
+    table_should_have_column,
+    column_should_be_timelike
+)
+import sqlparse
+
+
+class FromObj(object):
+    def __init__(self, from_obj, name, knowledge_date_column):
+        self.from_obj = from_obj
+        self.name = name
+        self.knowledge_date_column = knowledge_date_column
+
+    @property
+    def table(self):
+        if self.should_materialize():
+            return self.materialized_table
+        else:
+            return self.from_obj
+
+    @property
+    def materialized_table(self):
+        return f"{self.name}_from_obj"
+
+    @property
+    def create_materialized_table_sql(self):
+        return f"create table {self.materialized_table} as (select * from {self.from_obj})"
+
+    @property
+    def index_materialized_table_sql(self):
+        return f"create index on {self.materialized_table} ({self.knowledge_date_column})"
+
+    @property
+    def drop_materialized_table_sql(self):
+        return f"drop table if exists {self.materialized_table}"
+
+    def should_materialize(self):
+        statements = sqlparse.parse(self.from_obj)
+        # sqlparse.parse returns a tuple of Statement objects. a from_obj should have just one
+        if len(statements) > 1:
+            raise ValueError(f"Expecting only one statement in from_obj {self.from_obj}")
+        statement = statements[0]
```

Comment:
not really a substantive difference, but just to note, I like optimizing/organizing around the optimal case (doing first and asking questions later) --

```
try:
    (statement,) = sqlparse.parse(self.from_obj)
except ValueError as exc:
    raise ValueError("my message") from exc
```

(plus this catches the perhaps unusual case that that list is empty.)